packages/sign-client/src/controllers/engine.ts

```diff
@@ -2941,4 +2974,56 @@ export class Engine extends IEngine {
       }
     }
   };
+
+  private shouldSetTVF = (
+    protocolMethod: JsonRpcTypes.WcMethod,
+    params: JsonRpcTypes.RequestParams["wc_sessionRequest"],
+  ) => {
+    if (!params) return false;
+    if (protocolMethod !== "wc_sessionRequest") return false;
+    const { request, chainId } = params;
+    const namespace = parseChainId(chainId).namespace;
+    if (!(namespace in TVF_METHODS)) return false;
+    const tvfMethods = TVF_METHODS[namespace as keyof typeof TVF_METHODS];
+    if (!tvfMethods) return false;
+    return tvfMethods.includes(request.method);
+  };
+
+  private getTVFParams = (
+    id: number,
+    params: JsonRpcTypes.RequestParams["wc_sessionRequest"],
+    txHashes?: string[],
```

Comment:
how do you set txHashes? can you point to the logic that parses these caluse for Solana RPCs?

Reply:
Check: https://github.com/reown-com/reown-kotlin/pull/46/files#diff-d87d8f12338566a38f75a24780006abad01dd5865be0e803b842c36be0c5012aR33

Reply:
updated!